### PR TITLE
Amend execution unwind logic

### LINKED ISSUE #528: Unwind execution logic wrong when unwind_to == 0
Due to [this code](https://github.com/torquem-ch/silkworm/blob/master/node/silkworm/stagedsync/stage_execution.cpp#L233-L237) it happens that when unwinding to block 0 (corner case) all records for table `PlainState` get deleted and this implies there is no more state generated by genesis allocations.

### PR DESCRIPTION
**Pull request by AndreaLanfranchi**
Fixes #528

### CODE REVIEW

File: node/silkworm/stagedsync/stage_execution.cpp [hunk 1]
```diff
@@ -278,31 +276,28 @@ StageResult unwind_execution(db::RWTxn& txn, const std::filesystem::path&, uint6
...
+        // Note erasing forward the start key is included that's why we increase unwind_to by 1
+        Bytes start_key(8, '\0');
+        endian::store_big_u64(&start_key[0], unwind_to + 1);
+        for (const auto& map_config : unwind_tables) {
+            auto unwind_cursor{db::open_cursor(*txn, map_config)};
+            auto erased{db::cursor_erase(unwind_cursor, start_key, db::CursorMoveDirection::Forward)};
+            if(erased > 16) {
+                log::Info() << "Erased " << erased << " records from " << map_config.name;
             }
+            unwind_cursor.close();
```
Comment (yperbasis) [inline node/silkworm/stagedsync/stage_execution.cpp:300]: Will be closed automatically by the destructor, won't it?

File: node/silkworm/stagedsync/stage_execution.cpp [hunk 2]
```diff
@@ -278,31 +276,28 @@ StageResult unwind_execution(db::RWTxn& txn, const std::filesystem::path&, uint6
...
-            // Delete records which has keys greater than unwind point
-            // Note erasing forward the start key is included that's why we increase unwind_to by 1
-            Bytes start_key(8, '\0');
-            endian::store_big_u64(&start_key[0], unwind_to + 1);
-            for (int i = 2; i < 7; ++i) {
-                auto unwind_cursor{db::open_cursor(*txn, unwind_tables[i])};
-                auto erased{db::cursor_erase(unwind_cursor, start_key, db::CursorMoveDirection::Forward)};
-                log::Info() << "Erased " << erased << " records from " << unwind_tables[i].name;
-                unwind_cursor.close();
+        {
```
Comment (yperbasis) [inline node/silkworm/stagedsync/stage_execution.cpp:280]: Why leave the braces?

**Review by yperbasis — APPROVED** [review]

File: node/silkworm/stagedsync/stage_execution.cpp (hunk 2 above)
Comment (AndreaLanfranchi) [reply node/silkworm/stagedsync/stage_execution.cpp:280]: So opened cursors get closed as they fall out of scope

File: node/silkworm/stagedsync/stage_execution.cpp (hunk 1 above)
Comment (AndreaLanfranchi) [reply node/silkworm/stagedsync/stage_execution.cpp:300]: Yeah right

**Pull request merged**